node/node.go

```diff
@@ -819,6 +819,53 @@ func (node *AlgorandFullNode) oldKeyDeletionThread() {
 
 		r := node.ledger.Latest()
 
+		// We need the latest header to determine the next compact cert
+		// round, if any.
+		latestHdr, err := node.ledger.BlockHdr(r)
+		if err != nil {
+			switch err.(type) {
+			case ledger.ErrNoEntry:
+				// No need to warn; expected during catchup.
+			default:
+				node.log.Warnf("Cannot look up latest block %d for deleting ephemeral keys: %v", r, err)
+			}
+			continue
+		}
+		latestProto := config.Consensus[latestHdr.CurrentProtocol]
+
+		// If compact certs are enabled, we need to determine what signatures
+		// we already computed, since we can then delete ephemeral keys that
+		// were already used to compute a signature (stored in the compact
+		// cert db).
+		ccSigs, err := node.compactCert.LatestSigsFromThisNode()
+		if err != nil {
+			node.log.Warnf("Cannot look up latest compact cert sigs: %v", err)
+			continue
+		}
+
+		nextRoundFunc := func(part account.Participation) basics.Round {
```

Comment:
passing the nextRoundFunc to the DeleteOldKeys is technically correct, but makes it much harder to follow.
Could you have that implementation as part of the DeleteOldKeys or in  `accountManager.go` ?

Reply:
Yeah, it's not the cleanest code..  The reason I tried doing it this way (with a callback func) is to avoid having the lower layers depend on knowing about compact certificates.  In particular, currently the compactcert package imports the data package, and we would need to import the compactcert package from data if we wanted to invoke `node.compactCert.LatestSigsFromThisNode()` there.

Just for comparison, I pushed another version of this code: `node.go` calls `LatestSigsFromThisNode()`, but then the callback is inlined inside of `accountManager.go`.  If this looks cleaner to you, I'm happy with it too.